stubs/resources/js/Components/Menu/AppMenu.vue

```diff
@@ -1,16 +1,67 @@
 <template>
     <div>
-        <ul v-for="(item, index) in items" :key="index">
+        <li class="mb-2 block">
+            <div class="rounded-tl rounded-tr">
+                <label for="search" class="sr-only">Search</label>
+                <div class="flex items-center align-middle">
+                    <div
+                        class="pointer-events-none absolute flex items-center pl-3"
+                    >
+                        <i class="ri-search-line"></i>
+                    </div>
+                    <AppInputText
+                        id="search"
+                        v-model="searchTerm"
+                        :placeholder="__('Search...')"
+                        class="w-full pl-9"
+                    ></AppInputText>
+
+                    <AppButton
+                        v-if="searchTerm"
+                        class="btn ml-1 border border-skin-neutral-8 bg-skin-neutral-5 hover:bg-skin-neutral-8"
+                        @click="searchTerm = ''"
+                    >
+                        <i class="ri-close-line"></i>
+                    </AppButton>
+                </div>
+            </div>
+        </li>
+
+        <ul v-for="(item, index) in filteredItems" :key="index">
             <AppMenuSection :item="item" />
         </ul>
     </div>
 </template>
 
 <script setup>
-defineProps({
+import { ref } from 'vue'
+import { computed } from 'vue'
+
+const props = defineProps({
     items: {
         type: Array,
         default: () => []
     }
 })
+
+const searchTerm = ref('')
+
+// Computed property to filter items based on search term
+const filteredItems = computed(() => {
+    if (!searchTerm.value) return props.items
+    const searchFilter = (item) => {
+        // Convert search term to lowercase for case-insensitive matching
+        const term = searchTerm.value.toLowerCase()
+
+        // Check if the label or any child item contains the search term
+        const matches = (i) =>
+            i.label?.toLowerCase().includes(term) ||
+            (i.children && i.children.some(matches)) // Recursively check children
+
+        return matches(item)
+    }
+
+    // Filter items recursively
+    return props.items.filter(searchFilter)
+})
```

Comment:
The idea of a search can be useful. I made some tests and updated it to cover other scenarios so for tests purposes:

`resources/js/Configs/menu.js`
```js
export default {
    // main navigation - side menu
    items: [
        {
            label: 'Dashboard',
            permission: 'Dashboard',
            icon: 'ri-dashboard-line',
            link: route('dashboard.index')
        },

        {
            label: 'Banners',
            permission: 'Main Menu: Banners',
            children: [
                {
                    label: 'Sections',
                    permission: 'Main Menu: Banners: Sections - List',
                    icon: 'ri-landscape-line',
                    link: route('dashboard.index')
                }
            ]
        },

        {
            label: 'Page',
            permission: 'Page',
            children: [
                {
                    label: 'Pages',
                    permission: 'Page: Page - List',
                    icon: 'ri-draft-line',
                    link: route('dashboard.index')
                }
            ]
        },

        {
            label: 'Blog',
            permission: 'Blog',
            children: [
                {
                    label: 'Posts',
                    permission: 'Blog: Post - List',
                    icon: 'ri-draft-line',
                    link: route('dashboard.index')
                },
                {
                    label: 'Categories',
                    permission: 'Blog: Category - List',
                    icon: 'ri-folders-line',
                    link: route('dashboard.index')
                },
                {
                    label: 'Tags',
                    permission: 'Blog: Tag - List',
                    icon: 'ri-price-tag-3-line',
                    link: route('dashboard.index')
                },
                {
                    label: 'Authors',
                    permission: 'Blog: Author - List',
                    icon: 'ri-team-line',
                    link: route('dashboard.index')
                }
            ]
        },

        {
            label: 'Newsletter',
            permission: 'Newsletter',
            children: [
                {
                    label: 'Newsletter List',
                    permission: 'Newsletter: List',
                    icon: 'ri-mail-line',
                    link: route('dashboard.index')
                }
            ]
        },

        {
            label: 'Access Control List',
            permission: 'Acl',
            children: [
                {
                    label: 'Users',
                    permission: 'Acl: User - List',
                    icon: 'ri-user-line',
                    link: route('user.index')
                },
                {
                    label: 'Permissions',
                    permission: 'Acl: Permission - List',
                    icon: 'ri-shield-keyhole-line',
                    link: route('aclPermission.index')
                },
                {
                    label: 'Roles',
                    permission: 'Acl: Role - List',
                    icon: 'ri-account-box-line',
                    link: route('aclRole.index')
                }
            ]
        }
    ]
}
```

Updated search:

```js
import { ref, computed } from 'vue'

const props = defineProps({
    items: {
        type: Array,
        default: () => []
    }
})

const searchTerm = ref('')

// Computed property to filter items based on search term
const filteredItems = computed(() => {
    if (!searchTerm.value) return props.items
    const term = searchTerm.value.toLowerCase()

    // Recursive function to filter items and their children
    const filterItems = (items) => {
        return items.reduce((acc, item) => {
            const isParentMatch = item.label.toLowerCase().includes(term)
            let filteredChildren = []

            if (item.children) {
                if (isParentMatch) {
                    // If parent matches, include all children
                    filteredChildren = item.children
                } else {
                    // Else, filter children based on search term
                    filteredChildren = filterItems(item.children)
                }
            }

            // Include the item if the parent matches or any of its children match
            if (
                isParentMatch ||
                (filteredChildren && filteredChildren.length)
            ) {
                acc.push({
                    ...item,
                    // If parent matches, include all children; else, include filtered children
                    children: isParentMatch ? item.children : filteredChildren
                })
            }

            return acc
        }, [])
    }

    return filterItems(props.items)
})
```
Now it will return the expected values for a larger amount of scenarios like:
"Access Control List" => It's a parent node. Return the parent and all childs.
"Posts" => It's a child node. Returns the parent label "Blog" and the searched term "Posts".

Please update to use this concept...